e2e-test/cromwell_azure_e2e_test.py

```diff
@@ -0,0 +1,161 @@
+            if current_running_workflow_count == 0:
+                logging.info("Workflow(s) finished running")
+            else:
+                # Reset current count to 0 for next retry
+                # Decrement max_retries by 1
+                # Wait for sleep_timer before checking workflow statuses again (adjust value as needed)
+                logging.info(f"These workflows have yet to return a completed status: [{', '.join(workflow_ids)}]")
+                max_retries -= 1
+                current_running_workflow_count = 0
+                time.sleep(sleep_timer)
+    logging.info("Workflow(s) submission and completion successful")
+        
+def test_cleanup(workspace_namespace, workspace_name, bearer_token):
+    try:
+        delete_workspace(workspace_namespace, workspace_name, rawls_url, bearer_token)
+        logging.info("Workspace cleanup complete")
+    # Catch the exeception and continue with the test since we don't want cleanup to affect the test results
+    # We can assume that Janitor will clean up the workspace if the test fails
+    except Exception as e:
+        logging.info("Error cleaning up workspace, test script will continue")
+        logging.info(f'Exception details below:\n{e}')
+        
+def main():
+    workspace_name = ""
+    workspace_id = ""
+    found_exception = False
+    
+    # Sleep timers for various steps in the test
+    provision_sleep_timer = 60 * 5
+    workflow_run_sleep_timer = 60 * 5
+    workflow_status_sleep_timer = 60 * 2
+    
+    try:
+        (workspace_id, workspace_name) = create_workspace(billing_project_name, token, rawls_url)
+
+        create_app(workspace_id, leo_url, 'WORKFLOWS_APP', 'WORKSPACE_SHARED', token)
+        workflow_app_url = poll_for_app_url(workspace_id, 'WORKFLOWS_APP', 'cbas', token, leo_url)
+        
+        if not workflow_app_url:
+            raise Exception("Prerequisite workflow app URL could not be found.")
+        
+        create_app(workspace_id, leo_url, 'CROMWELL_RUNNER_APP', 'USER_PRIVATE', token)
+        app_url = poll_for_app_url(workspace_id, 'CROMWELL_RUNNER_APP', 'cromwell-runner', token, leo_url)
+
+        if not app_url:
+            raise Exception("App URL could not be found.")
+
+        # This chunk of code only executes one workflow
+        # Would like to modify this down the road to execute and store references for multiple workflows
+        workflow_response = submit_hello_world_to_cromwell(app_url, "Run Workflow Test", token)
+        logging.info(f'Executing sleep for {workflow_run_sleep_timer} seconds to allow workflow(s) to complete')
+        time.sleep(workflow_run_sleep_timer)
+
+        # This chunk of code supports checking one or more workflows
+        # Probably won't require too much modification if we want to run additional submission tests
+        workflow_ids = deque([workflow_response['id']])
+        get_completed_workflow(app_url, workflow_ids, token, sleep_timer=workflow_status_sleep_timer)
```

Comment:
Should the status and metadata of workflow be obtained from `cromwell-reader` from `WORKFLOWS` app to recreate the scenario that happens in UI? Or does the UI directly talk to Cromwell Runner for workflow status? 🤔

Reply:
That's a good question - I'd like to leave it as-is because this test is trying to check the Cromwell -> TES -> Azure Batch workflow execution pipeline as narrowly as possible. We're only starting the WORKFLOWS app up at all because it's required to create the metadata database. We might want to have a different test that does more to mimic the UI, though.